src/dataflow/src/sink/kafka.rs

```diff
@@ -698,6 +703,9 @@ where
     >| {
         if s.shutdown_flag.load(Ordering::SeqCst) {
             info!("shutting down sink: {}", &s.name);
+            // Indicate that the sink is closed to everyone else who
+            // might be tracking its write frontier.
+            s.write_frontier.borrow_mut().clear();
```

Comment:
this is duplicative because theres only one path where we end up in `SendState::Shutdown` but I think 4 or so paths where we set the `shutdown_flag` to true

Reply:
Does going into the shutdown state not cause this path to be hit on the next iteration?

Reply:
there's some paths that set the flag, but not the state, e.g. if we asynchronously discover that we failed to produce to the broker

```
 219     fn delivery(&self, result: &DeliveryResult, _: Self::DeliveryOpaque) {
 220         match result {
 221             Ok(_) => (),
 222             Err((e, msg)) => {
 223                 self.metrics.message_delivery_errors_counter.inc();
 224                 error!(
 225                     "received error while writing to kafka sink topic {}: {}",
 226                     msg.topic(),
 227                     e
 228                 );
 229                 self.shutdown_flag.store(true, Ordering::SeqCst);
 230             }
 231         }
 232     }
 ```

Reply:
I meant the other way around - is there a problem with only clearing the write frontier here and not in the shutdown state?

Reply:
ah i see. yeah i think thats fine

Reply:
Should I wait for that edit, or is this up for review in its current form?

Reply:
i made the edit! sorry just saw your question now

Reply:
oops - this was a brain fart where I thought I had made the edit but hadn't. actually pushed the edit now!